packages/viewer/src/modules/tree/RenderTree.ts

```diff
@@ -29,10 +29,13 @@ export class RenderTree {
     this.root = subtreeRoot
   }
 
-  public buildRenderTree(geometryConverter: GeometryConverter): Promise<boolean> {
+  public buildRenderTree(
+    geometryConverter: GeometryConverter,
+    callback: () => void
+  ): Promise<boolean> {
     const p = this.tree.walkAsync((node: TreeNode): boolean => {
       let start = performance.now()
-      const rendeNode = this.buildRenderNode(node, geometryConverter)
+      const rendeNode = this.buildRenderNode(node, geometryConverter, callback)
```

Comment:
You'll need to be careful what this callback does. Large streams have hundreds of thousands of nodes that get built

Reply:
It's mean to just be a callback for the event and calculation.  I don't readily see a different way to ensure it's not used for nefarious things.